app/models/dashboard.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require "uri"
+require "net/http"
+require "httparty"
+require "json"
+require "openssl"
+
+class Dashboard
+  include HTTParty
+  base_uri ENV["GF_BASE_URL"]
+
+  attr_reader :program, :gf_dashboard
+
+  def initialize(program)
+    @program = program
+    @gf_dashboard = program.gf_dashboard || program.build_gf_dashboard
+  end
+
+  def create
+    create_org
+    switch_to_current_org
+    create_org_token
+    create_datasource
+    create_dashboard
+    set_default_dashboard
+  end
+
+  def create_org
+    option = { basic_auth: auth, body: { name: program.name } }
+
+    result = self.class.post("/api/orgs", option)
+    gf_dashboard.update(org_id: result.parsed_response["orgId"]) if result.response.is_a?(Net::HTTPSuccess)
+  end
+
+  def switch_to_current_org
+    option = { basic_auth: auth, body: {} }
+
+    self.class.post("/api/user/using/#{gf_dashboard.org_id}", option)
+  end
+
+  def create_org_token
+    option = { basic_auth: auth, body: { name: "csc_apikey", role: "Admin" } }
```

Comment:
@hengsokly what is the name: "csc_apikey" used for? please review to ensure that key/access token won't leak